tests/profile/tracking/test_tracking_sessions.py

```diff
@@ -0,0 +1,280 @@
+    open_session = format_session(
+        'active-session', 987654321, os.getpid(), 55, True
+    )
+    sample_sessions.append(open_session)
+    setup_sessions_dir(sample_sessions)
+    setup_paused_sessions_file(None)
+
+    listed_sessions = tracking_sessions.get_open_sessions()
+
+    for session in listed_sessions:
+        assert session.is_open()
+
+    assert len(listed_sessions) == 1
+
+    listed_session = listed_sessions[0]
+
+    open_session_path = tracking_sessions.get_session_file_path(
+        open_session['name'],
+        open_session['pid']
+    )
+
+    assert os.path.basename(open_session_path) == listed_session.file
+    assert os.path.abspath(open_session_path) == listed_session.path
+    assert open_session['name'] == listed_session.name
+    assert open_session['pid'] == listed_session.pid
+
+
+@pytest.mark.parametrize('name, pid, expected', [
+    ('test-1', 1234, os.path.join(tracker_dir, '1234-test-1.json')),
+    ('test-2', 5830, os.path.join(tracker_dir, '5830-test-2.json')),
+])
+def test_get_session_file_path(name, pid, expected):
+    session_file_path = os.path.abspath(
+        tracking_sessions.get_session_file_path(name, pid)
+    )
+    assert session_file_path == expected
+
+
+# @pytest.mark.parametrize('name, pid', TEST_DATA)
+# def test_get_session_unique_id(name, pid):
+#     pass
+
+
+def test_session_start_with_pid():
+    setup_sessions_dir([])
+    setup_paused_sessions_file(None)
+
+    test_session = TrackingSession(name='test', pid=1234)
+    session_path = tracking_sessions.session_start(
+        test_session.name, test_session.pid
+    )
+
+    assert os.path.abspath(session_path) == test_session.path
+
+    with open(test_session.path, 'r') as test_session_f:
+        session_data = json.load(test_session_f)
+
+    assert test_session.name == session_data['name']
+    assert test_session.pid == session_data['pid']
+
+
+def test_session_start_no_pid():
+    setup_sessions_dir([])
+    setup_paused_sessions_file(None)
+
+    test_session = TrackingSession(name='test', pid=os.getpid())
+    session_path = tracking_sessions.session_start(
+        test_session.name
+    )
+
+    assert os.path.abspath(session_path) == test_session.path
+
+    with open(test_session.path, 'r') as test_session_f:
+        session_data = json.load(test_session_f)
+
+    assert test_session.name == session_data['name']
+    assert test_session.pid == session_data['pid']
+
+
+def test_session_end():
+    pass
+
+
+def test_get_paused_sessions():
+    setup_paused_sessions_file(SAMPLE_SESSIONS)
+
+    paused_sessions = tracking_sessions.get_paused_sessions()
+
+    assert len(paused_sessions) == len(SAMPLE_SESSIONS)
+
+    for session in SAMPLE_SESSIONS:
+        session_obj = TrackingSession(name=session['name'], pid=session['pid'])
+        assert session_obj in paused_sessions
+
+
+# @pytest.mark.parametrize('session', TEST_DATA)
+# def test_pause_tracking_session(session):
```

Comment:
There is a `test_pause_tracking_sessions`, so do you still need this one? (same with the one below)

Reply:
I've removed this.